platform/security/core/security-core-impl/src/main/java/ddf/security/samlp/impl/LogoutMessageImpl.java

```diff
@@ -122,8 +129,15 @@ public LogoutRequest extractSamlLogoutRequest(@NotNull String samlLogoutResponse
   @Override
   public LogoutRequest buildLogoutRequest(
       @NotNull String nameIdString, @NotNull String issuerOrEntityId, List<String> sessionIndexes) {
-    return buildLogoutRequest(
-        nameIdString, issuerOrEntityId, getUuidGenerator().generateUuid(), sessionIndexes);
+    return buildLogoutRequest(nameIdString, issuerOrEntityId, generateUuid(), sessionIndexes);
+  }
+
+  public String generateUuid() {
+    if (Objects.isNull(getUuidGenerator())) {
```

Comment:
We shouldn't need to have the runtime null check here. We generally rely on the assumption that blueprint will always give us a non null property (We should always get a proxy object that will eventually resolve to the real service)

Reply:
I added it because I had originally misconfigured my blueprint file. The ultimate results was that clicking "Logout" on the Logout UI gave you the message. "You have been logged out", but if you clicked "sign in again", it would take you back to your application without asking for a username/password. 

That scared me, so I made sure that the logger would alert an administrator of (potentially) severe security situation. 

Options are: 

1. Take out null check entirely.
2. Take out the RuntimeException, but continue to log the error. (A null pointer exception gets logged at INFO level).
3. Leave as-is.

Reply:
Talked with Ryan. This is really a more general issue. Created a new ticket  to address the general case. DIB-7658.